alpha/lib/model/LiveEntry.php

```diff
@@ -629,80 +642,80 @@ public function setMediaServer($index, $hostname, $applicationName = null)
 
 		KalturaLog::debug("about to setMediaServer. index: $index, hostname: $hostname");
 		$this->setLastBroadcast(kApiCache::getTime());
-		$server = new kLiveMediaServer($index, $hostname, $mediaServerNode->getDc(), $mediaServerNode->getId(), $applicationName ? $applicationName : $mediaServerNode->getApplicationName());
-		$this->putInCustomData("server-$index", $server, LiveEntry::CUSTOM_DATA_NAMESPACE_MEDIA_SERVERS);
-		$this->setLiveStatus(LiveEntryStatus::PLAYABLE, $index);
+		$dbLiveEntryServerNode->setStatus(EntryServerNodeStatus::PLAYABLE);
+		$dbLiveEntryServerNode->setServerType($index);
+		$dbLiveEntryServerNode->setServerNodeId($mediaServerNode->getId());
+		$dbLiveEntryServerNode->save();
+		$this->setLiveStatus(EntryServerNodeStatus::PLAYABLE, $index);
 	}
-	
+
+	public function unsetMediaServer($index)
+	{
+		// Set media server with index #index to be stopped
+		// the entry->liveStatus() is calculated from the state of both servers
+		$this->setLiveStatus(EntryServerNodeStatus::STOPPED, $index);
+	}
+
+
+	private function getEntryServerNodeCacheKey(EntryServerNode $entryServerNode)
+	{
+		return $entryServerNode->getId()."_".$entryServerNode->getServerType()."_".$entryServerNode->getServerNodeId();
+	}
+
 	protected function isMediaServerRegistered($index, $hostname)
 	{
-		$server = $this->getFromCustomData("server-$index", LiveEntry::CUSTOM_DATA_NAMESPACE_MEDIA_SERVERS);
-		/* @var $server kLiveMediaServer */
-		if($server && $server->getHostname() == $hostname)
+		/* @var $dbLiveEntryServerNode LiveEntryServerNode*/
```

Comment:
Please validate hostName matches and also validate entryServerNode is on status PLAYABLE as in my previous comments.